addons/mass_mailing/models/mass_mailing.py

```diff
@@ -57,6 +59,67 @@ def _compute_contact_nbr(self):
         for mailing_list in self:
             mailing_list.contact_nbr = data.get(mailing_list.id, 0)
 
+    @api.multi
+    def action_merge(self, src_lists, archive):
+        """
+            Insert all the contact from the mailing lists 'src_lists' to the
+            mailing list 'self'. Archive the mailing lists 'src_lists'
+            except the destination mailing list 'self'.
+        """
+
+        # Explation of the SQL query with an example. There are the following lists
```

Comment:
Update comment / docstring to explain row_number notably, because it is the magic trick of this method that is interesting.